### PR TITLE
 Added support for NXS2 blue waveform previews and color waveforms 

### PR DESCRIPTION
**Pull request by jan2000**
Added support for NXS2 waveforms.

```
$ ./monitor-qt.py -h
usage: monitor-qt.py [-h] [--disable-pdb] [--nxs2-blue] [--nxs2-color]

Python ProDJ Link

optional arguments:
  -h, --help     show this help message and exit
  --disable-pdb  Disable PDB provider
  --nxs2-blue    Show NXS2 blue waveform previews
  --nxs2-color   Show NXS2 color waveforms
```

image

### CODE REVIEW

**Comment by brunchboy** [comment]
You figured it out, that’s wonderful! 🎉 🎊

File: packets.py [hunk 1]
```diff
@@ -490,6 +490,8 @@ def _sizeof(self, context, path):
   unknown1_request = 0x2504, # issued when loading track, reply 0x4502, contains lots of 0 and some data at end
   waveform_request = 0x2904,
   unknown2_request = 0x2b04, # issued when loading track, reply 0x4e02, no render request
+  color_preview_waveform_request = 0x2c04, # nxs2 ext request, seen with PWV4, PWV5, PVB2 PQT2
+  color_waveform_request = 0x2c04,
```
Comment (brunchboy) [inline packets.py:494]: Both of these requests can’t use the same value, `0x2c04`, surely?

File: packets.py [hunk 2]
```diff
@@ -503,7 +505,9 @@ def _sizeof(self, context, path):
   beatgrid = 0x4602,
   cues = 0x4702,
   waveform = 0x4a02,
-  unknown2 = 0x4e02
+  unknown2 = 0x4e02,
+  color_preview_waveform = 0x4f02, # nxs2 ext response
+  color_waveform = 0x4f02,
```
Comment (brunchboy) [inline packets.py:510]: Similarly, the preview and scrolling waveform probably use different responses, not both sharing `0x4f02`?

**Comment by flesniak** [comment]
Wow, that looks like a huge overhaul! Thanks a lot. I will try to take a lot into it this evening.

File: packets.py (hunk 1 above)
Comment (brunchboy) [reply packets.py:494]: Ah! Never mind, I found an nxs2 packet capture someone had sent me, and now understand what you are saying: the `2c04` message is a general request for an EXT file segment, and its arguments identify the rekordbox ID and tag type desired. Interesting!

**Comment by flesniak** [comment]
Finally managed to merge this one, thanks a lot for the amazing work.
I changed some minor details as merging old and colored widgets to support displaying blue waveforms when no colored ones are available. I changed the command line arguments - which I should have introduced long ago, so thanks a lot for that as well - to a bit more descriptive names.

**Pull request closed without merging**